### ISSUE TITLE
Out-of-bounds crash in `igraph_spanner()`

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2487]
**Describe the bug**

Out-of-bounds crash in `igraph_spanner()`, found with fuzzing.

**To reproduce**

Run with address sanitizer:

```c
    igraph_t graph;
    igraph_vector_int_t spanner;

    igraph_vector_int_init(&spanner, 0);
    igraph_rng_seed(igraph_rng_default(), 42);
    igraph_small(&graph, 2, IGRAPH_DIRECTED, 0, 1, -1); // P_2 graph, this is connected
    igraph_spanner(&graph, &spanner, 1.72, NULL);
    igraph_destroy(&graph);
    igraph_vector_int_destroy(&spanner);
```

See this branch: https://github.com/igraph/igraph/tree/tests/igraph_spanner-crash

CI failure on this branch: https://dev.azure.com/igraph-team/igraph/_build/results?buildId=6433&view=logs&j=4c3a8381-b9de-5463-904f-918810769be8&t=e71559a0-54d3-55f3-64cc-74b67ad70398&l=563

```
==66526==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x0001202190f8 at pc 0x00010255bb2c bp 0x00016da3e490 sp 0x00016da3e488
READ of size 8 at 0x0001202190f8 thread T0
    #0 0x10255bb28 in igraph_i_collect_lightest_edges_to_clusters sparsifier.c:81
    #1 0x10255b2f8 in igraph_spanner sparsifier.c:425
    #2 0x1023cd9e0 in main igraph_spanner.c:195
    #3 0x1827d50dc  (<unknown module>)

0x0001202190f8 is located 8 bytes before 336-byte region [0x000120219100,0x000120219250)
allocated by thread T0 here:
    #0 0x103d63610 in wrap_calloc+0x9c (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x53610)
    #1 0x10245dfa8 in igraph_vector_init vector.pmt:144
    #2 0x1025592a0 in igraph_spanner sparsifier.c:230
    #3 0x1023cd9e0 in main igraph_spanner.c:195
    #4 0x1827d50dc  (<unknown module>)
```

**Version information**
Current `master`

**Comment by szhorvat** [comment]
@guyroznb, would you have time to take a look at this?

**Comment by szhorvat** [comment]
The documentation claimed that edge directions are ignored, but this was not the case. I fixed this in 910d6e79b1277d4944a7ebe422dcd35bd40a2cd1

@guyroznb The documentation also claims that the input graph should be connected. Is this a hard requirement? Do we need to check for this?
